src/backend/renderer.js

```diff
@@ -672,6 +688,26 @@ export function attach(
     pendingOperations.length = 0;
     pendingRealUnmountedIDs.length = 0;
     pendingSimulatedUnmountedIDs.length = 0;
+    pendingStringTable.clear();
+    pendingStringTableLength = 0;
+    pendingStringTableCounter = 0;
+  }
+
+  function getStringID(str: string | null): number {
+    if (str === null) {
+      return 0;
+    }
+    const existingID = pendingStringTable.get(str);
+    if (existingID !== undefined) {
+      return existingID;
+    }
+    let id = ++pendingStringTableCounter;
```

Comment:
Why do we need a separate `pendingStringTableCounter` variable instead of just using `pendingStringTable.size`?

Reply:
Good catch. Fixed.

Reply:
Hmm. Let me think more about this for a minute.

Reply:
Yeah that should be good.